docs/src/components/Contributor.astro

```diff
@@ -1,59 +1,75 @@
 ---
 import { Icon } from '@astrojs/starlight/components';
 
-interface Props {
+interface Contributor {
 	name: string;
 	twitter?: string;
 	linkedin?: string;
 	github?: string;
 	website?: string;
 }
 
-const { name, twitter, linkedin, github, website } = Astro.props;
+interface Props {
+	contributors: Contributor[];
+}
+
+const { contributors } = Astro.props;
 ---
 
-<p class="contributor">
-	Created by {name}
-	{
-		twitter && (
-			<a href={twitter}>
-				<Icon class="icon" name="twitter" size="0.75rem" />
-			</a>
-		)
-	}
-	{
-		linkedin && (
-			<a href={linkedin}>
-				<Icon class="icon" name="linkedin" size="0.75rem" />
-			</a>
-		)
-	}
-	{
-		github && (
-			<a href={github}>
-				<Icon class="icon" name="github" size="0.75rem" />
-			</a>
-		)
-	}
+<p class="contributors">
+	Created by
+
 	{
-		website && (
-			<a href={website}>
-				<Icon class="icon" name="external" size="0.75rem" />
-			</a>
+		contributors.map(
+			({ data: { name, twitter, linkedin, github, website } }) => (
```

Comment:
Minor/question: Do we need to mention `data` in the type?

`contributors` is supposed to have `Contributor[]` type, but seems to have `{ data: Contributor[] }` instead. Is this something specific to Astro?

Reply:
Using 'getEntry' to get a contributor from the collection returns the data in this format:

```
  {
    id: 'chau-tran',
    collection: 'contributors',
    data: {
      name: 'Chau Tran',
      twitter: 'https://twitter.com/Nartc1410',
      github: 'https://github.com/nartc'
    }
  }
```

(the component is passed an array of these, which is why I am mapping over it and destructuring on data)

I don't actually use Astro so I'm not sure if there is a nicer way to go about it or not

Reply:
(typing from phone so can't test this myself) - Not 100% sure, but I think the type _could_ use something like this (copied from one of my repos):

```ts
import type { CollectionEntry } from 'astro:content';
// interface Props {
// 	 page: Page<CollectionEntry<'contributor'>>;
// }
// OR
interface Props {
	contributors: Array<CollectionEntry<'contributor'>>;
}
```